components/derive/src/service_factory.rs

```diff
@@ -42,6 +44,55 @@ fn check_artifact_name(name: impl AsRef<[u8]>) -> bool {
     name.as_ref().iter().copied().all(is_allowed_latin1_char)
 }
 
+#[derive(Debug, Default)]
+struct AdditionalInterfaces(Vec<Path>);
+
+impl FromMeta for AdditionalInterfaces {
+    fn from_string(value: &str) -> darling::Result<Self> {
+        Path::from_string(value).map(|path| Self(vec![path]))
+    }
+
+    fn from_value(value: &Lit) -> darling::Result<Self> {
+        Path::from_value(value).map(|path| Self(vec![path]))
+    }
+
+    fn from_list(items: &[NestedMeta]) -> darling::Result<Self> {
+        items
+            .iter()
+            .map(|meta| match meta {
+                NestedMeta::Lit(lit) => Path::from_value(lit),
+                _ => Err(darling::Error::unsupported_format(
+                    "Additional services should be in format: `additional(\"First\", \"Second\")`",
+                )),
+            })
+            .collect::<Result<Vec<_>, _>>()
+            .map(AdditionalInterfaces)
+    }
+}
+
+#[derive(Debug, FromMeta)]
+struct ServiceInterfaces {
+    default: Path,
+    #[darling(default)]
+    additional: AdditionalInterfaces,
+}
+
+impl ServiceInterfaces {
+    fn list_all(&self) -> impl IntoIterator<Item = (String, &Path)> {
+        // Only the additional interfaces use the explicit interface names.
+        let additional_interfaces = self.additional.0.iter().map(|path| {
+            let interface_name = path
+                .segments
+                .last()
```

Comment:
Nit: It could make sense to make `interface_name`s more fool-proof:

- The imported name can be changed compared to the declared name of the interface
- Simple name can lead to interface name collisions

Protobuf-like qualified names (e.g., `exonum.Token`) could solve collisions, but I'm not sure how to pass these names to the proc macro in a satisfactory manner. So far, I've come up with terrible crutches:

```rust
/// This is the service interface.
trait ServiceInterface {
    // methods...
}

/// This is the descriptor interface defined in the core.
trait DescribeInterface {
    const NAME: &'static str;
}

/// Automatically generated via `exonum_service` proc macro.
/// Requires `ServiceInterface` to be object-safe.
impl DescribeInterface for dyn ServiceInterface {
    const NAME: &'static str = "foo.ServiceInterface";
}

// We can then determine interface name as:
println!("{}", <dyn ServiceInterface as DescribeInterface>::NAME);
```